[azure] bring your own data example

kristapratico opened the pull request:
Adds a notebook showing how to access the Azure chat completion models with your own data (preview) with the Python SDK.

kristapratico commented:
@ted-at-openai this PR is ready - would you mind reviewing when you get the chance? Thanks!

simonpfish commented on examples/azure/chat_with_your_own_data.ipynb:
```diff
@@ -0,0 +1,432 @@
+    "import openai\n",
+    "import dotenv\n",
+    "\n",
+    "dotenv.load_dotenv()"
```
would be good to specify with a bit more detail what should go into the dotenv and maybe do it all in one step (as opposed to also talking about openai env variables below). 

an alternative approach could be to use something like `input` or `getpass` to prompt the user to input the variables

kristapratico replied:
Good point - I've included more detail on what needs to be added for dotenv

simonpfish commented on examples/azure/chat_with_your_own_data.ipynb:
```diff
@@ -0,0 +1,432 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Providing our search endpoint, key, and index name for the `dataSources` keyword argument, any questions posed to the model will now be grounded in your own data. An additional property, `context`, will be provided to show the data the model referenced to answer the question."
```
to me it is not very clear what data this example is using. it would be great if we could provide an easy to use example endpoint, or more information about what is actually going under the hood here

simonpfish reviewed (commented):
thank you for the contribution. bring your own data looks like a valuable offering.

left a couple comments, once those are resolved you should be good to go.

simonpfish reviewed (changes requested):
passing along to your queue

kristapratico commented:
> passing along to your queue

@simonpfish thanks a lot for the detailed review! I believe I've addressed your feedback, but let me know if there is anything else I can clarify.

simonpfish commented:
Great! Thank you for addressing the feedback! 

One last bit, could you add something in the introduction to place this in context: how does this compare to other methods for data retrieval? is this something a developer could do on their own? what are some alternative methods to follow?
